pandas/tests/typing/valid/test_frame.py

```diff
@@ -0,0 +1,835 @@
+
+
+def test_types_setitem() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4], 5: [6, 7]})
+    i = pd.Index(["col1", "col2"])
+    s = pd.Series(["col1", "col2"])
+    a = np.array(["col1", "col2"])
+    df["col1"] = [1, 2]
+    df[5] = [5, 6]
+    df[["col1", "col2"]] = [[1, 2], [3, 4]]
+    df[s] = [5, 6]
+    df[a] = [[1, 2], [3, 4]]
+    df[i] = [8, 9]
+
+
+def test_types_setitem_mask() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4], 5: [6, 7]})
+    select_df = pd.DataFrame({"col1": [True, True], "col2": [False, True]})
+    df[select_df] = [1, 2, 3]
+
+
+def test_types_iloc_iat() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df.iloc[1, 1]
+    df.iloc[[1], [1]]
+    df.iat[0, 0]
+
+
+def test_types_loc_at() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df.loc[[0], "col1"]
+    df.at[0, "col1"]
+
+
+def test_types_boolean_indexing() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df[df > 1]
+    df[~(df > 1.0)]
+
+
+def test_types_head_tail() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df.head(1)
+    df.tail(1)
+
+
+def test_types_assign() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df.assign(col3=lambda frame: frame.sum(axis=1))
+    df["col3"] = df.sum(axis=1)
+
+
+def test_types_sample() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df.sample(frac=0.5)
+    df.sample(n=1)
+
+
+def test_types_nlargest_nsmallest() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df.nlargest(1, "col1")
+    df.nsmallest(1, "col2")
+
+
+def test_types_filter() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df.filter(items=["col1"])
+    df.filter(regex="co.*")
+    df.filter(like="1")
+
+
+def test_types_setting() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    df["col1"] = 1
+    df[df == 1] = 7
+
+
+def test_types_drop() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4]})
+    res: pd.DataFrame = df.drop("col1", axis=1)
+    res2: pd.DataFrame = df.drop(columns=["col1"])
+    res3: pd.DataFrame = df.drop({0})
+    res4: pd.DataFrame = df.drop(index={0})
+    res5: pd.DataFrame = df.drop(columns={"col1"})
+    res6: pd.DataFrame = df.drop(index=1)
+    res7: pd.DataFrame = df.drop(labels=0)
+    res8: None = df.drop([0, 0], inplace=True)
+
+
+def test_types_dropna() -> None:
+    df = pd.DataFrame(data={"col1": [np.nan, np.nan], "col2": [3, np.nan]})
+    res: pd.DataFrame = df.dropna()
+    res2: pd.DataFrame = df.dropna(axis=1, thresh=1)
+    res3: None = df.dropna(axis=0, how="all", subset=["col1"], inplace=True)
+
+
+def test_types_fillna() -> None:
+    df = pd.DataFrame(data={"col1": [np.nan, np.nan], "col2": [3, np.nan]})
+    res: pd.DataFrame = df.fillna(0)
+    res2: None = df.fillna(method="pad", axis=1, inplace=True)
```

Comment:
This fails with a `NotImplementedError` during "Test experimental data manager" (it works on the other CI runs).

@jbrockmendel

Reply:
This looks correct, but could use a better exception message (maybe something other than NotImplementedError).  In this particular case, we don't have an effective way of actually respecting `inplace=True`